parachains/runtimes/collectives/collectives-polkadot/src/impls.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright (C) 2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use frame_support::{
+	dispatch::{DispatchError, DispatchResultWithPostInfo},
+	log,
+	traits::{Currency, Get, Imbalance, OnUnbalanced, OriginTrait},
+	weights::Weight,
+};
+use pallet_alliance::{ProposalIndex, ProposalProvider};
+use sp_std::{boxed::Box, marker::PhantomData};
+use xcm::latest::{Fungibility, Junction, NetworkId, Parent};
+
+type AccountIdOf<T> = <T as frame_system::Config>::AccountId;
+
+type ProposalOf<T, I> = <T as pallet_collective::Config<I>>::Proposal;
+
+type HashOf<T> = <T as frame_system::Config>::Hash;
+
+type NegativeImbalanceOf<T, I> = <<T as pallet_alliance::Config<I>>::Currency as Currency<
+	<T as frame_system::Config>::AccountId,
+>>::NegativeImbalance;
+
+type CurrencyOf<T, I> = <T as pallet_alliance::Config<I>>::Currency;
+
+type BalanceOf<T, I> = <<T as pallet_alliance::Config<I>>::Currency as Currency<
+	<T as frame_system::Config>::AccountId,
+>>::Balance;
+
+/// Implements `OnUnbalanced::on_unbalanced` to teleport slashed assets to relay chain treasury account.
+pub struct ToParentTreasury<TreasuryAcc, TempAcc, T, I = ()>(
+	PhantomData<(TreasuryAcc, TempAcc, T, I)>,
+);
+
+impl<TreasuryAcc, TempAcc, T, I: 'static> OnUnbalanced<NegativeImbalanceOf<T, I>>
+	for ToParentTreasury<TreasuryAcc, TempAcc, T, I>
+where
+	TreasuryAcc: Get<AccountIdOf<T>>,
+	TempAcc: Get<AccountIdOf<T>>,
```

Comment:
If this is going to be `SLASHED_IMBALANCE_ACC_ID`, I don't think it is needed to make it a configurable generic anymore, but as you wish.